test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_String.java

```diff
@@ -1860,6 +1860,18 @@ public void test_join2() {
 		}
 	}
 
+	/**
+	 * @tests java.lang.String#replaceAll(String, String)
+	 */
+	@Test
+	public void test_replaceAll() {
+		String replace1 = "1a2a3a";
+		String result1 = replace1.replaceAll("\u1161", "[");
+		AssertJUnit.assertTrue("replaceAll() compact result should be identical", replace1 == result1);
+		String replace2 = "1a2b3c\u1161";
+		String result2 = replace2.replaceAll("\u1162", "\u1234");
+		AssertJUnit.assertTrue("replaceAll() non-compact result should be identical", replace1 == result1);
+	}
```

Comment:
I think this test is sensitive to big- vs. little-endian (the character `\u1161` is stored `0x61`, `0x11` on little-endian systems, but `0x11`, `0x61` on big-endian systems (where it won't fail even without the fix). Adding one or more `\u0011` characters to `replace1` and `replace2` should address that.

Reply:
Ok. I think it's only needed for replace1 so I only added it there.

Reply:
On big-endian systems `0x11` is the value that's significant for both tests. If `replace2` doesn't contain that byte, it's not really testing the problem.

Reply:
The only "problem" for non-compact strings (i.e. replace2) is that it didn't return the identical string when nothing was replaced. For non-compact string it looks at chars, not bytes.